[addons][filesystem][network] fix mem crash on filesystem and add user agent ask to network

AlwinEsch opened the pull request:
## Description

Patch 1:
---------------------------
**[addons][filesystem] fix missing set to nullptr**

There was on new functions a char* value used, where Kodi gives his string (strdup) to.
But if not available the the value was not set, as previous also no nullptr set, has it tried to "free" it and makes a crash.

Patch 2:
---------------------------
**[addons][network] add function to get Kodi's network user agent**

The new function is "kodi::network::GetUserAgent()" where return e.g.
- **"Kodi/19.0-ALPHA1 (X11; Linux x86_64) Ubuntu/20.04 App_Bitness/64 Version/19.0-ALPHA1-Git:20200522-0076d136d3-dirty"**

<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
With modified screensaver addon to call them.
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

AlwinEsch committed:
[addons][network] add function to get Kodi's network user agent

The new function is "kodi::network::GetUserAgent()" where return e.g.
"Kodi/19.0-ALPHA1 (X11; Linux x86_64) Ubuntu/20.04 App_Bitness/64 Version/19.0-ALPHA1-Git:20200522-0076d136d3-dirty"

phunkyfish reviewed (approved):
Looks good 👍

garbear commented:
There are other opportunities to fix missing initializers, like [here](https://github.com/xbmc/xbmc/blob/0076d136d31cb100b31b47a6fe728c162d6a0e24/xbmc/addons/kodi-addon-dev-kit/include/kodi/Filesystem.h#L2227)

AlwinEsch committed:
[addons][filesystem] fix missing set to nullptr

There was on new functions a char* value used, where Kodi gives his string (strdup) to.
But if not available the the value was not set, as previous also no nullptr set, has
it tried to "free" it and makes a crash.

AlwinEsch commented:
Thanks @garbear found there two places where was not set and is now there.
